### PR TITLE
Add WIP Readme

### LINKED ISSUE #5: add docs
add documentation for developers and users on how to use/setup this plugin.

### PR DESCRIPTION
**Pull request by Akin909**
Add an initial `README` explaining how this plugin works. Closes #5

### CODE REVIEW

**Comment by Akin909** [comment]
@kyazdani42 I've added some docs to the readme and a brief intro didn't really do a super detailed project info since I'm short on time. Hopefully this helps for now

File: README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+### Setup
+
+This adds all the highlight groups for the devicons
+i.e. it calls `highlight IconNameDevicon guifg='<color>'` for all icons
+this might need to be re-called in a `Colorscheme` to re-apply cleared highlights
+if the color scheme changes
+
+```lua
+require'nvim-lua-devicons'.setup()
```
Comment (kyazdani42) [inline README.md:26]: nvim-web-devicons

File: README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+### Get Icon
+
+Get the icon for a given file by passing in the `name`, the `extension` and an _optional_ options `table`.
+The name is passed in to check for an exact match e.g. `.bashrc` if there is no exact name match the extension
+is used.
+
+```lua
+require'nvim-lua-devicons'.get_icon(filename, extension, options)
```
Comment (kyazdani42) [inline README.md:36]: nvim-web-devicons

File: README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+The optional `options` argument can used to change how the plugin works the keys include
+`default = <boolean>`. If the default key is set to true this function will return a default
+if there is no matching icon
+e.g.
+
+```lua
+require'nvim-lua-devicons'.get_icon(filename, extension, { default = true })
```
Comment (kyazdani42) [inline README.md:45]: nvim-web-devicons x)

Comment (Akin909) [reply README.md]: Fixed :+1:

**Pull request merged**